### PR TITLE
cache path added

### PR DESCRIPTION
**Pull request by t-nanava**
**Setting Cache path from Environment**

#### Description
The location to the cache file formed by V8 engine is provided as an environment variable.

#### Validation
The current branch is tested with V8 engine and validated with the help of RNTester application.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native/pull/98)

### CODE REVIEW

File: ReactCommon/cxxreact/V8Executor.cpp [hunk 1]
```diff
@@ -212,6 +212,23 @@ Isolate *V8Executor::GetIsolate() {
...
 }
 
+/*
+* This below function used incase user set the enviorment variable
+*  to enable cache insted of enable from C++ reacthost instance.
+*/
+void V8Executor::TrySetJSEConfigFromEnv(){
+  const char* localPath = getenv("V8_JSCACHE_PATH");
+  if (localPath != NULL) {
+    if ( m_jseConfigParams == nullptr ) {
```
Comment (rohitjain85) [inline ReactCommon/cxxreact/V8Executor.cpp]: >  [](start = 8, length = 1)

nit: extra space here and after nullptr

File: ReactCommon/cxxreact/V8Executor.cpp [hunk 2]
```diff
@@ -212,6 +212,23 @@ Isolate *V8Executor::GetIsolate() {
...
+      LOGV("V8Executor.setCacheFromEnv V8_JSCACHE_PATH: %s", localPath);
+      m_jseConfigParams = std::make_unique<facebook::react::JSEConfigParams>();
+      m_jseConfigParams->cachePath = std::string(localPath);
+      m_jseConfigParams->cacheType = CachingType::FullCachingWithNoLazy;
+      m_jseConfigParams->loggingLevel = 50; // deafult logging 10 = Logging::LoggingLevel::ERROR
```
Comment (rohitjain85) [inline ReactCommon/cxxreact/V8Executor.cpp]: >ERROR [](start = 91, length = 5)

what 50 stands for, INFO?

File: ReactCommon/cxxreact/V8Executor.cpp [hunk 3]
```diff
@@ -247,6 +264,9 @@ V8Executor::V8Executor(std::shared_ptr<ExecutorDelegate> delegate,
   m_jscConfig(jscConfig),
   m_jseConfigParams(std::move(jseConfigParams)) {
   LOGV("V8Executor::V8Executor entry");
+  if ( m_jseConfigParams == nullptr ) {
```
Comment (rohitjain85) [inline ReactCommon/cxxreact/V8Executor.cpp]: >  [](start = 6, length = 1)

nit: extra space here and after nullptr

File: ReactCommon/cxxreact/V8Executor.cpp (hunk 2 above)
Comment (t-nanava) [reply ReactCommon/cxxreact/V8Executor.cpp]: Yes

**Pull request merged**